_examples/kafka-to-http/router/main.go

```diff
@@ -0,0 +1,69 @@
+package main
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/ThreeDotsLabs/watermill"
+	"github.com/ThreeDotsLabs/watermill/message"
+	watermill_http "github.com/ThreeDotsLabs/watermill/message/infrastructure/http"
+	"github.com/ThreeDotsLabs/watermill/message/infrastructure/kafka"
+	"github.com/ThreeDotsLabs/watermill/message/router/plugin"
+)
+
+var (
+	brokers   = []string{"kafka:9092"}
+	logger    = watermill.NewStdLogger(false, false)
+	topicName = "kafka_to_http_example"
+)
+
+// filterMessages passes the message along if its event type is one of acceptedTypes.
+func filterMessages(acceptedTypes ...string) message.HandlerFunc {
+	return func(msg *message.Message) ([]*message.Message, error) {
+		// the kafka producer sets this metadata so that we don't have to unmarshal the body
+		// just sort the messages based on event type metadata
+		msgEventType := msg.Metadata.Get("event_type")
+
+		for _, typ := range acceptedTypes {
+			if string(typ) == msgEventType {
+				return message.Messages{msg}, nil
+			}
+		}
+
+		return nil, nil
+	}
+}
+
+func marshalMessage(topic string, msg *message.Message) (*http.Request, error) {
```

Comment:
what you think about just removing this function, using DefaultMarshalMessageFunc and url in `router.AddHandler()`, it will be much more clear

Reply:
82f0556f36e29d8788da5984ab4a37909ac5e8c2